code/LeftAndMain.php

```diff
@@ -386,29 +388,76 @@ public function schema($request)
         return $this->getSchemaResponse($schemaID, $form);
     }
 
+    /**
+     * Return an error HTTPResponse encoded as json
+     *
+     * @param int $errorCode
+     * @param string $errorMessage
+     * @return HTTPResponse
+     * @throws HTTPResponse_Exception
+     */
+    public function jsonError($errorCode, $errorMessage = null)
+    {
+        // Build error from message
+        $error = [
+            'type' => 'error',
+            'code' => $errorCode,
+        ];
+        if ($errorMessage) {
+            $error['value'] = $errorMessage;
+        }
+
+        // Support explicit error handling with status = error, or generic message handling
+        // with a message of type = error
+        $result = [
+            'status' => 'error',
```

Comment:
I think this `'status'` doesn't do anything and isn't used anywhere currently, but it's a good start for an explicit error handle